features/facebookImport/test/ministory/advertising-value-analysis.test.js

```diff
@@ -0,0 +1,97 @@
+import AdvertisingValueAnalysis from "../../src/model/analyses/ministories/advertising-value-analysis";
+import FacebookAccount from "../../src/model/entities/facebook-account";
+import { runAnalysisForAccount } from "../utils/analyses-execution";
+import {
+    expectInactiveAnalysis,
+    expectAnalysisSuccessStatus,
+    expectActiveAnalysis,
+} from "../utils/analysis-assertions";
+
+describe("Your advertising value analysis from account with no ad interests", () => {
+    let analysis = null;
+    let status = null;
+
+    beforeAll(async () => {
+        const facebookAccount = new FacebookAccount();
+        ({ analysis, status } = await runAnalysisForAccount(
+            AdvertisingValueAnalysis,
+            facebookAccount
+        ));
+    });
+
+    it("has success status", async () => {
+        expectAnalysisSuccessStatus(status);
+    });
+
+    it("is not active", async () => {
+        expectInactiveAnalysis(analysis);
+    });
+});
+
+const DATASETS = [
+    {
+        adInterests: ["Interest"],
+        numberInterests: 1,
+        randomAdInterestsCount: 1,
+    },
+    {
+        adInterests: ["InterestOne", "InterestTwo"],
+        numberInterests: 2,
+        randomAdInterestsCount: 2,
+    },
+    {
+        adInterests: ["InterestOne", "InterestTwo", "InterestThree"],
+        numberInterests: 3,
+        randomAdInterestsCount: 3,
+    },
+
+    {
+        adInterests: [
+            "InterestOne",
+            "InterestTwo",
+            "InterestThree",
+            "InterestFour",
+        ],
+        numberInterests: 4,
+        randomAdInterestsCount: 3,
+    },
+];
+for (const dataset of DATASETS) {
+    describe("Your advertising value analysis from account with no ad interests", () => {
+        let analysis = null;
+        let status = null;
+
+        beforeAll(async () => {
+            let facebookAccount = new FacebookAccount();
```

Comment:
The problem seems to be that there's no easy way to create an account with just the data that we need... Or to test the data decoupled from the facebookAccount, is that correct?

Reply:
We can always introduced a mocked account if using directly `FacebookAccount` will work in some cases. Until now we either manually create a `FacebookAccount` and place relevant data in it, or start from a zip file and get the account by importing the zip